File: packages/pumpkin-ts-models/src/models/MediaAsset.ts
export type MediaAssetLicenseStatus =
  | 'unknown'
  | 'needs_review'
  | 'approved'
  | 'rejected'
  | 'owned'
  | 'licensed'
  | 'ai_generated'
  | 'partner_provided';

export type MediaAssetUsageStatus =
  | 'unused'
  | 'in_use'
  | 'needs_review'
  | 'approved_for_publish';

export type MediaAssetStatus =
  | 'draft'
  | 'active'
  | 'archived'
  | 'replaced'
  | 'deleted-pending';

export type MediaAssetUsageType =
  | 'hero'
  | 'card'
  | 'gallery'
  | 'og-image'
  | 'icon'
  | 'background'
  | 'inline'
  | 'document';

export type MediaAssetStorageProvider =
  | 'local-dev'
  | 'azure-blob'
  | 'external';

export interface MediaAssetFocalPoint {
  x: number | null;
  y: number | null;
}

export interface MediaAssetUsageReference {
  pageId: string;
  pageSlug: string;
  fieldPath: string;
  blockType: string;
  imageRole: string;
}

export interface MediaAssetVariant {
  name: string;
  url: string;
  publicUrl: string;
  width: number | null;
  height: number | null;
  mimeType: string;
  sizeBytes: number | null;
  storageProvider: MediaAssetStorageProvider;
  blobPath: string;
  generatedAt: string;
  status: 'available' | 'planned' | 'failed';
}

export interface MediaAsset {
  id: string;
  tenantId: string;
  siteKey?: string;
  assetId: string;
  status?: MediaAssetStatus;
  url: string;
  publicUrl?: string;
  thumbnailUrl?: string;
  fileName: string;
  originalFileName?: string;
  safeFileName?: string;
  title: string;
  alt: string;
  altText?: string;
  caption: string;
  source: string;
  credit?: string;
  license?: string;
  sourceUrl: string;
  usageType?: MediaAssetUsageType;
  licenseStatus: MediaAssetLicenseStatus;
  usageStatus: MediaAssetUsageStatus;
  width: number | null;
  height: number | null;
  mimeType: string;
  extension?: string;
  fileSize: number | null;
  sizeBytes?: number | null;
  checksum?: string;
  hash?: string;
  storageProvider?: MediaAssetStorageProvider;
  storageContainer?: string;
  blobPath?: string;
  focalPoint: MediaAssetFocalPoint;
  decorative: boolean;
  tags: string[];
  notes: string;
  variants?: MediaAssetVariant[];
  createdAt: string;
  updatedAt: string;
  createdBy: string;
  uploadedBy?: string;
  lastReviewedAt: string;
  reviewedBy: string;
  usageReferences: MediaAssetUsageReference[];
  usedByPages?: MediaAssetUsageReference[];
  replacedByMediaAssetId?: string;
  archivedAt?: string;
  archivedBy?: string;
}

File: apps/pumpkin-net-models/Models/MediaAsset.cs
using System.Text.Json.Serialization;

namespace pumpkin_net_models.Models;

public class MediaAsset
{
    [JsonPropertyName("id")]
    public string Id { get; set; } = string.Empty;

    [JsonPropertyName("tenantId")]
    public string TenantId { get; set; } = string.Empty;

    [JsonPropertyName("siteKey")]
    public string SiteKey { get; set; } = string.Empty;

    [JsonPropertyName("assetId")]
    public string AssetId { get; set; } = string.Empty;

    [JsonPropertyName("status")]
    public string Status { get; set; } = "draft";

    [JsonPropertyName("url")]
    public string Url { get; set; } = string.Empty;

    [JsonPropertyName("publicUrl")]
    public string PublicUrl { get; set; } = string.Empty;

    [JsonPropertyName("thumbnailUrl")]
    public string ThumbnailUrl { get; set; } = string.Empty;

    [JsonPropertyName("fileName")]
    public string FileName { get; set; } = string.Empty;

    [JsonPropertyName("originalFileName")]
    public string OriginalFileName { get; set; } = string.Empty;

    [JsonPropertyName("safeFileName")]
    public string SafeFileName { get; set; } = string.Empty;

    [JsonPropertyName("title")]
    public string Title { get; set; } = string.Empty;

    [JsonPropertyName("alt")]
    public string Alt { get; set; } = string.Empty;

    [JsonPropertyName("altText")]
    public string AltText { get; set; } = string.Empty;

    [JsonPropertyName("caption")]
    public string Caption { get; set; } = string.Empty;

    [JsonPropertyName("source")]
    public string Source { get; set; } = string.Empty;

    [JsonPropertyName("credit")]
    public string Credit { get; set; } = string.Empty;

    [JsonPropertyName("license")]
    public string License { get; set; } = string.Empty;

    [JsonPropertyName("sourceUrl")]
    public string SourceUrl { get; set; } = string.Empty;

    [JsonPropertyName("usageType")]
    public string UsageType { get; set; } = "inline";

    [JsonPropertyName("licenseStatus")]
    public string LicenseStatus { get; set; } = "unknown";

    [JsonPropertyName("usageStatus")]
    public string UsageStatus { get; set; } = "unused";

    [JsonPropertyName("width")]
    public int? Width { get; set; }

    [JsonPropertyName("height")]
    public int? Height { get; set; }

    [JsonPropertyName("mimeType")]
    public string MimeType { get; set; } = string.Empty;

    [JsonPropertyName("extension")]
    public string Extension { get; set; } = string.Empty;

    [JsonPropertyName("fileSize")]
    public long? FileSize { get; set; }

    [JsonPropertyName("sizeBytes")]
    public long? SizeBytes { get; set; }

    [JsonPropertyName("checksum")]
    public string Checksum { get; set; } = string.Empty;

    [JsonPropertyName("hash")]
    public string Hash { get; set; } = string.Empty;

    [JsonPropertyName("storageProvider")]
    public string StorageProvider { get; set; } = "external";

    [JsonPropertyName("storageContainer")]
    public string StorageContainer { get; set; } = string.Empty;

    [JsonPropertyName("blobPath")]
    public string BlobPath { get; set; } = string.Empty;

    [JsonPropertyName("focalPoint")]
    public MediaAssetFocalPoint FocalPoint { get; set; } = new();

    [JsonPropertyName("decorative")]
    public bool Decorative { get; set; } = false;

    [JsonPropertyName("tags")]
    public List<string> Tags { get; set; } = new();

    [JsonPropertyName("notes")]
    public string Notes { get; set; } = string.Empty;

    [JsonPropertyName("variants")]
    public List<MediaAssetVariant> Variants { get; set; } = new();

    [JsonPropertyName("createdAt")]
    public string CreatedAt { get; set; } = string.Empty;

    [JsonPropertyName("updatedAt")]
    public string UpdatedAt { get; set; } = string.Empty;

    [JsonPropertyName("createdBy")]
    public string CreatedBy { get; set; } = string.Empty;

    [JsonPropertyName("uploadedBy")]
    public string UploadedBy { get; set; } = string.Empty;

    [JsonPropertyName("lastReviewedAt")]
    public string LastReviewedAt { get; set; } = string.Empty;

    [JsonPropertyName("reviewedBy")]
    public string ReviewedBy { get; set; } = string.Empty;

    [JsonPropertyName("usageReferences")]
    public List<MediaAssetUsageReference> UsageReferences { get; set; } = new();

    [JsonPropertyName("usedByPages")]
    public List<MediaAssetUsageReference> UsedByPages { get; set; } = new();

    [JsonPropertyName("replacedByMediaAssetId")]
    public string ReplacedByMediaAssetId { get; set; } = string.Empty;

    [JsonPropertyName("archivedAt")]
    public string ArchivedAt { get; set; } = string.Empty;

    [JsonPropertyName("archivedBy")]
    public string ArchivedBy { get; set; } = string.Empty;
}

public class MediaAssetFocalPoint
{
    [JsonPropertyName("x")]
    public double? X { get; set; }

    [JsonPropertyName("y")]
    public double? Y { get; set; }
}

public class MediaAssetUsageReference
{
    [JsonPropertyName("pageId")]
    public string PageId { get; set; } = string.Empty;

    [JsonPropertyName("pageSlug")]
    public string PageSlug { get; set; } = string.Empty;

    [JsonPropertyName("fieldPath")]
    public string FieldPath { get; set; } = string.Empty;

    [JsonPropertyName("blockType")]
    public string BlockType { get; set; } = string.Empty;

    [JsonPropertyName("imageRole")]
    public string ImageRole { get; set; } = string.Empty;
}

public class MediaAssetVariant
{
    [JsonPropertyName("name")]
    public string Name { get; set; } = string.Empty;

    [JsonPropertyName("url")]
    public string Url { get; set; } = string.Empty;

    [JsonPropertyName("publicUrl")]
    public string PublicUrl { get; set; } = string.Empty;

    [JsonPropertyName("width")]
    public int? Width { get; set; }

    [JsonPropertyName("height")]
    public int? Height { get; set; }

    [JsonPropertyName("mimeType")]
    public string MimeType { get; set; } = string.Empty;

    [JsonPropertyName("sizeBytes")]
    public long? SizeBytes { get; set; }

    [JsonPropertyName("storageProvider")]
    public string StorageProvider { get; set; } = string.Empty;

    [JsonPropertyName("blobPath")]
    public string BlobPath { get; set; } = string.Empty;

    [JsonPropertyName("generatedAt")]
    public string GeneratedAt { get; set; } = string.Empty;

    [JsonPropertyName("status")]
    public string Status { get; set; } = "available";
}
